analysis/filter/EdgeFilter/src/main/kotlin/de/maibornwolff/codecharta/filter/edgefilter/EdgeProjectBuilder.kt

```diff
@@ -60,9 +59,13 @@ class EdgeProjectBuilder(private val project: Project, private val pathSeparator
     private fun insertEdgeAsNode(nodeEdgeName: String) {
         val nodeFilename = nodeEdgeName.split(pathSeparator).reversed().first()
         val nodePath = nodeEdgeName.split(pathSeparator)
-        val nodeParentPath = nodePath.subList(2, max(2, nodePath.size - 1))
-        val node = Node(nodeFilename, NodeType.File)
-        insertNodeInProjectBuilder(node, nodeParentPath)
+        if (nodePath.size <= 1) {
+            println("No parent node found")
```

Comment:
Im not sure if we should display this message, because it is printed at the command line without any information which edge is affected. So maybe the specified message is a bit confusing, isn't it? What do you think?

Reply:
I am not sure either as I didn't add this change. And honestly I don't know why it's there. It has nothing to do with the CLI.

Reply:
This is something @Hall-Ma and @polinasch ran into during the initial development in tests. If I recall correct it's about a faulty cc structure / path unrelated to each other. In that case, the bug would be hidden. It should probably already throw an error before reaching this point by doing some sanity checks. To at least let the user know something is wrong, this has been added. The message itself should be improved and we might even throw an error here (I am not certain how the tool is used, it might be caught upper layers).

Reply:
I removed the println and log an error instead. Furthermore, I improved the error message a bit.